### ISSUE TITLE
[pigeon] Java generator swaps code and message in exceptions

### ISSUE THREAD

**Issue by blastrock** [flutter/flutter#124393]
## Steps to Reproduce

1. Use Pigeon's java generator with `--java_out`
2. Throw an exception in Java/Kotlin code
3. Catch it in dart code
 
**Expected results:** `code` should be the name of the class and `message` the `.toString()`

**Actual results:** They are inverted

The bug is right here https://github.com/flutter/packages/blob/main/packages/pigeon/lib/java_generator.dart#L803-L810 . The `if` part is correct, the `else` is not.

**Comment by stuartmorgan** [comment]
This was discussed when the `FlutterError` handling was added, but it looks like it didn't get filed. We should consider what exactly we want the behavior to be here; the corresponding MethodChannel code is [here](https://github.com/flutter/engine/blob/3e9edddf7425719e9fac67890c30fb0ab1b1326c/shell/platform/android/io/flutter/plugin/common/MethodChannel.java#L277-L281), and it uses `getMessage` rather than `toString` as the message, which is probably a better choice. Using the class as the code is probably better than using "error" though.

The issue here is that this is a very subtle breaking change: everyone's code will still compile, but *clients of the plugins* will get different errors.

@tarrinneal We should probably pair this with another breaking change to Java to increase the chances that people with Java code will see the warning in the CHANGELOG.

**Comment by stuartmorgan** [comment]
(I think in earlier discussions I was leaning toward not fixing this due to the subtle break, but having thought about it more, having it be wrong forever for all future use is the worse option.)

**Comment by blastrock** [comment]
Thanks for the quick answer!

About having the class name as `code`, it would indeed break code. But I'm fairly sure that the code in our project worked before and was broken because of this. Before [this patch](https://github.com/flutter/packages/commit/5662a7e579#diff-3569133ccf0ec5f1bb08e4a8b2b9688913c7104783d2dc788ccbe685da3e4c2aL745-R748) the class name was put in the error code if I understand correctly, so this might be a regression.

**Comment by stuartmorgan** [comment]
Yes, it looks like that was indeed an accidental regression, where it used the order the keys happened to have been listed in for the array instead of the correct values. Unfortunately we didn't have test coverage of the specific error fields. (cc @tarrinneal FYI)

**Comment by stuartmorgan** [comment]
I just noticed that Swift is backwards too. We should sweep all the generators; maybe we need to do a single breaking change across multiple generators :(

**Comment by tarrinneal** [comment]
Yeah, one of the downsides of this being my first real project within this codebase. I didn't realize the code correlated to a Flutter Error, so I didn't think the order was important. I just kept it the same as before.
